src/Models/AZ3166Device.ts

```diff
@@ -160,93 +241,17 @@ export class AZ3166Device extends ArduinoDeviceBase {
-          await this.generateCrc(this.extensionContext, this.channel);
-      return retValue;
-    } else if (configSelection.detail === 'Config Connection String') {
-      // Get IoT Hub device connection string from config
-      let deviceConnectionString =
-          ConfigHandler.get<string>(ConfigKey.iotHubDeviceConnectionString);
-      const deviceConnectionStringSelection =
-          this.getDeviceConnectionStringOptions(deviceConnectionString);
-      const selection = await vscode.window.showQuickPick(
-          deviceConnectionStringSelection,
-          {ignoreFocusOut: true, placeHolder: 'Choose an option:'});
-      if (!selection) {
-        return false;
-      }
-
-      if (selection.label === 'Input IoT Hub Device Connection String') {
-        const option = this.getInputDeviceConnectionStringOptions();
-        deviceConnectionString = await vscode.window.showInputBox(option);
-        if (!deviceConnectionString) {
-          const message =
-              'Need more information on how to get device connection string?';
-          const result: vscode.MessageItem|undefined =
-              await vscode.window.showWarningMessage(
-                  message, DialogResponses.yes, DialogResponses.no);
-          if (result === DialogResponses.yes) {
-            opn(constants.informationPageUrl);
-          }
-          return false;
-        }
-      }
-      if (!deviceConnectionString) {
-        return false;
-      }
-
-      console.log(deviceConnectionString);
-      const res = await this.setDeviceConfig(
-          deviceConnectionString, ConfigDeviceOptions.ConnectionString);
-      if (!res) {
-        return false;
-      } else {
-        vscode.window.showInformationMessage(
-            'Configure Device connection string completely.');
-        return true;
-      }
-    } else if (configSelection.detail === 'Config DPS') {
-      const option = this.getDPSConnectionStringOptions();
-      const deviceConnectionString = await vscode.window.showInputBox(option);
-      if (!deviceConnectionString) {
-        return false;
-      }
-
-      console.log(deviceConnectionString);
-      const res = await this.setDeviceConfig(
-          deviceConnectionString, ConfigDeviceOptions.DPS);
-      if (!res) {
-        return false;
-      } else {
-        vscode.window.showInformationMessage(
-            'Config DPS credentials completely.');
-        return true;
-      }
-
-    } else {
-      const option = this.getUDSStringOptions();
-      const UDS = await vscode.window.showInputBox(option);
-      if (!UDS) {
-        return false;
-      }
-
-      console.log(UDS);
-      const res = await this.setDeviceConfig(UDS, ConfigDeviceOptions.UDS);
-      if (!res) {
-        return false;
-      } else {
-        vscode.window.showInformationMessage(
-            'Configure Unique Device String (UDS) completed successfully.');
-        return true;
-      }
-    }
+    return configSelection.detail;

```

Comment:
> configSelection.detail [](start = 10, length = 23)

Suggest using 'label' of the selection instead of the 'details', which is usually a long description string of the selected option. 

Since you've define an Enum for the device setting options, you can convert the selection into the 'ConfigDeviceOptions' enum, and use enum instead of string in the subsequent steps

Reply:
Fixed.